kafka-cron/producer/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/Shopify/sarama"
+	"github.com/berkeli/kafka-cron/types"
+	"github.com/goccy/go-yaml"
+	"github.com/google/uuid"
+	"github.com/robfig/cron/v3"
+)
+
+var (
+	brokerList  = kingpin.Flag("brokerList", "List of brokers to connect").Default("localhost:9092").Strings()
+	topicPrefix = kingpin.Flag("topicPrefix", "Topic prefix, e.g. jobs will create topics for each cluster in the format jobs-cluster-a").Default("jobs").String()
+	cronPath    = kingpin.Flag("cronPath", "Path to cron file").Default("./cron.yaml").String()
+	partitions  = kingpin.Flag("partitions", "Number of partitions").Default("1").Int32()
+)
+
+func main() {
+	InitMonitoring(2112)
+
+	kingpin.Parse()
+	config := sarama.NewConfig()
+
+	admin, err := sarama.NewClusterAdmin(*brokerList, config)
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	config.Producer.RequiredAcks = sarama.WaitForAll
+	config.Producer.Return.Successes = true
+	producer, err := sarama.NewSyncProducer(*brokerList, config)
+	if err != nil {
+		log.Panic(err)
+	}
+	defer func() {
+		if err := producer.Close(); err != nil {
+			log.Panic(err)
+		}
+		if err := admin.Close(); err != nil {
+			log.Panic(err)
+		}
+	}()
+	cmds, err := ReadConfig(*cronPath)
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	err = CreateTopics(admin, cmds)
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	err = ScheduleJobs(producer, cmds)
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	sigs := make(chan os.Signal, 1)
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	<-sigs
+}
+
+func CreateTopics(admin sarama.ClusterAdmin, cmds []types.Command) error {
```

Comment:
So your approach here is to 
 * read all the commands from config file
 * create conceptual clusters and corresponding topics based on the commands
 
I'd be more inclined to define the known clusters separately in a specific config file, and then treat reference to any unknown cluster as a failure. What would happen if a user misspelled a cluster name in a config? You'd create new topics automagically but you'd have no consumers running for them, presumably.